### PR TITLE
Updated build scripts

### PR DESCRIPTION
**Pull request by thecodejunkie**
Updated the following

- Updated CAKE version to 0.17.0
- Updated `build.ps1` to use `Cake.CoreCLR` and to be more aligned with our other `build.ps1` files
- Updated `build.sh1` to use `Cake.CoreCLR`  and to be more aligned with our other `build.sh` files
- Updated `build.cake`
  - Removed dependency on `Newtonsoft.Json` for updating `project.json`
  - Massive formatting update - the formatting was all over the place and very inconsistent
  - Updated `ExecuteGit` to output the actual git instructions to make it easier to follow along the build progress

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,19 +1,36 @@
-# README
+# Description
 
-Release procedures:
+Cross-platform build script, using [CAKE](http://cakebuild.net/), which is used to release a new Nancy version. The script will update the main Nancy repository and all of the sub-repositories to the designed version and build & publish the Nuget packages.
 
-`./build.sh --target Prepare-Release --targetversion=X.X.X` - This will get all repos, set the Nancy version and push back to Nancy repo, update all sub repos with new Nancy version
+The build script can be invoke either with `build.sh` (maxOS/Linux) or `build.ps1` (Windows)
```
Comment (jchannon) [inline README.md]: Invoked

File: build.sh [hunk 2]
```diff
@@ -1,60 +1,64 @@
...
-    shift
-done
-
-function installnuget() {
-    echo "Checking for nuget"
-    if ! [ -x "$(command -v nuget)" ] ; then
-        echo "Installing nuget to $TOOLS_DIR"
-        wget -O $TOOLS_DIR $NUGET_URL
-        export PATH=$TOOLS_DIR:$PATH
+#!/usr/bin/env bash
```
Comment (jchannon) [inline build.sh:1]: Should this have -xe

File: build.sh [hunk 3]
```diff
@@ -1,60 +1,64 @@
...
+export PATH="$SCRIPT_DIR/.dotnet":$PATH
+export DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1
+export DOTNET_CLI_TELEMETRY_OPTOUT=1
+"$SCRIPT_DIR/.dotnet/dotnet" --info
+
+###########################################################################
+# INSTALL NuGet
+###########################################################################
+
+# if ! [ -x "$(command -v nuget)" ] ; then
```
Comment (jchannon) [inline build.sh]: Should this be commented out

**Review by jchannon — CHANGES_REQUESTED** [review]

File: build.sh (hunk 2 above)
Comment (thecodejunkie) [reply build.sh:1]: No it spits out a billion lines if we do 😄

File: build.sh (hunk 3 above)
Comment (thecodejunkie) [reply build.sh]: Good catch - that's going away

**Comment by thecodejunkie** [comment]
Pushed an update

**Comment by jchannon** [comment]
Won't cock your permissions up etc? The other repos have it

**Review by jchannon — APPROVED** [review]

**Comment by thecodejunkie** [comment]
No has nothing to do with permissions. It's about outputting the executed commands and exiting on the first error

**Pull request merged**